lib/qmi/codec/network_access.ex

```diff
@@ -149,6 +154,15 @@ defmodule QMI.Codec.NetworkAccess do
     parse_get_home_network_tlvs(parsed, rest)
   end
 
+  defp maybe_put_provider_name(home_network_resp, <<0x00>>) do
+    home_network_resp
+  end
+
+  defp maybe_put_provider_name(home_network_resp, <<length, rest::binary-size(length)>>) do
+    home_network_resp
+    |> Map.put(:provider, rest)
+  end
+
```

Comment:
What do you think about changing this to:

```suggestion
  defp extract_string(<<length, string::binary-size(length)>>), do: string
```

and then do a `|> Map.put(:provider, extract_string(description))` above. 

I like the idea that `:provider` will always be a key and if the description is an empty string, then you'll see an empty string.